package/boot/uboot-kirkwood/patches/160-sheevaplug.patch

```diff
@@ -0,0 +1,39 @@
+--- a/include/configs/sheevaplug.h
++++ b/include/configs/sheevaplug.h
+@@ -47,15 +47,17 @@
+ /*
+  * Default environment variables
+  */
+-#define CONFIG_BOOTCOMMAND		"${x_bootcmd_kernel}; "	\
+-	"setenv bootargs ${x_bootargs} ${x_bootargs_root}; "	\
+-	"bootm 0x6400000;"
++#define CONFIG_BOOTCOMMAND					\
++ "setenv bootargs ${console} ${mtdparts} ${bootargs_root}; "	\
++	"ubi part ubi; " 						\
++	"ubi read 0x800000 kernel; " 			\
++	"bootm 0x800000"
+ 
+-#define CONFIG_EXTRA_ENV_SETTINGS	"x_bootargs=console"	\
+-	"=ttyS0,115200 mtdparts="CONFIG_MTDPARTS_DEFAULT	\
+-	"x_bootcmd_kernel=nand read 0x6400000 0x100000 0x400000\0" \
+-	"x_bootcmd_usb=usb start\0" \
+-	"x_bootargs_root=root=/dev/mtdblock3 rw rootfstype=jffs2\0"
++#define CONFIG_EXTRA_ENV_SETTINGS			\
++	"console=console=ttyS0,115200\0"		\
++	"mtdids="CONFIG_MTDIDS_DEFAULT "\0"		\
++	"mtdparts="CONFIG_MTDPARTS_DEFAULT "\0"	\
++	"bootargs_root=\0"
+ 
+ /*
+  * Ethernet Driver configuration
+--- a/configs/sheevaplug_defconfig
++++ b/configs/sheevaplug_defconfig
+@@ -22,7 +22,7 @@
+ CONFIG_CMD_FAT=y
+ CONFIG_CMD_JFFS2=y
+ CONFIG_MTDIDS_DEFAULT="nand0=orion_nand"
+-CONFIG_MTDPARTS_DEFAULT="mtdparts=orion_nand:512K(uboot),512K(env),4M(kernel),-(rootfs)"
++CONFIG_MTDPARTS_DEFAULT="mtdparts=orion_nand:1M(uboot),-(ubi)"
```

Comment:
why not keep uboot and env separate here?

Reply:
Just my fault i just wanted to protect uboot+env.

Reply:
Oh and thanks for merging! And also for the 5.10 stuff!

Reply:
"fault" sounds like "wrong". If that's the case, consider fixing it in a separate PR ...

Reply:
Here is the same:
https://github.com/torvalds/linux/blob/cba5e97280f53ec7feb656fcdf0ec00a5c6dd539/arch/arm/boot/dts/kirkwood-sheevaplug-common.dtsi#L75

It could be corrected in new PR. Also entry for fw_printenv and setenv could be added.

IMO u-boot in kurkwood could be rw. This SoC is unbricable, if You have access to uart.